server/test/controllers/ErrorHandlerTest.java

```diff
@@ -0,0 +1,62 @@
+package controllers;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static play.mvc.Http.Status.BAD_REQUEST;
+import static play.mvc.Http.Status.INTERNAL_SERVER_ERROR;
+import static play.test.Helpers.fakeRequest;
+
+import controllers.admin.NotChangeableException;
+import org.junit.Before;
+import org.junit.Test;
+import play.mvc.Result;
+import repository.ResetPostgres;
+
+public class ErrorHandlerTest extends ResetPostgres {
+
+  private ErrorHandler handler;
+
+  @Before
+  public void setUpErrorHandler() {
+    handler = instanceOf(ErrorHandler.class);
+  }
+
+  @Test
+  public void findThrowableByType_findsNone() {
+    Throwable exception = new NotChangeableException("test exception");
+    assertThat(ErrorHandler.findThrowableByType(exception, IllegalStateException.class)).isEmpty();
+  }
+
+  @Test
+  public void findThrowableByType_findsSource() {
+    Throwable exception = new NotChangeableException("test exception");
+    assertThat(ErrorHandler.findThrowableByType(exception, RuntimeException.class))
+        .contains(exception);
+  }
+
+  @Test
+  public void findThrowableByType_findsNested() {
```

Comment:
Not blocking: Consider adding a test where the exception we want is nested >N levels deep and isn't found. The rationale being that it documents that there are limits to what the error handling can automatically detect.

Reply:
I'm not inclined to document exclusively through tests perse, I added it to the protected method doc though.

I think having the limit isn't a feature as much as a trade off so having a test fail with 6 levels of nesting doesn't seem useful either.